### PR TITLE
feat: add draggable list

### PR DESCRIPTION
**Pull request by tanvibhakta**
## Description

Zeplin: https://app.zeplin.io/project/63ac3e207fcd5a051ca87ddf/styleguide/components?seid=63db285d81ead948b4dacc3c
Fixes https://github.com/appsmithorg/appsmith/issues/19336

The designs in the storybook don't match the designs on zeplin because they will not be coming from here. This component is a wrapper around any component in the main repo that gives the child the ability to be dragged. 

## Type of change

- New feature (non-breaking change which adds functionality)

### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: packages/design-system/src/DraggableList/DraggableList.constants.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { CLASS_NAME_PREFIX } from "__config__/constants";
+
+export const DraggableListClassName = `${CLASS_NAME_PREFIX}-name`;
```
Comment (albinAppsmith) [inline packages/design-system/src/DraggableList/DraggableList.constants.ts:3]: This is template... Please update according to the component

File: packages/design-system/src/DraggableList/DraggableList.tsx [hunk 2]
```diff
@@ -0,0 +1,295 @@
+import React, { useRef, useEffect } from "react";
+import { useFocusRing } from "@react-aria/focus";
```
Comment (albinAppsmith) [inline packages/design-system/src/DraggableList/DraggableList.tsx:2]: remove this

File: packages/design-system/src/DraggableList/DraggableList.types.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
+export type DraggableListProps = {
+  /** this prop is used when... */
+  prop1: undefined;
```
Comment (albinAppsmith) [inline packages/design-system/src/DraggableList/DraggableList.types.ts:3]: ??

**Comment by tanvibhakta** [comment]
On hold because this is a snowflake.

**Pull request closed without merging**